lib/release.js

```diff
@@ -13,21 +13,18 @@ module.exports.createRelease = async function createRelease ({
   repoOwner,
   repoName
 }) {
-  const repo = { owner: repoOwner, repo: repoName }
-  const gh = new Octokit({ auth: gitHubToken })
+  const gh = new GitHubService(new Octokit({ auth: gitHubToken }), repoOwner, repoName)
 
-  const tagResponse = await gh.repos.listTags(repo)
-
-  const previousTag = semver
-    .sort(tagResponse.data.map(tag => tag.name))
-    .reverse()
-    .find(tag => semver.lt(tag, targetTag))
+  return createReleaseWithService(gh, targetTag)
+}
 
-  const comp = await gh.repos.compareCommits({
-    base: previousTag,
-    head: targetTag,
-    ...repo
-  })
+/**
+ * @param {import('./github').GitHubService} gh
+ * @param {string} targetTag
+ */
+async function createReleaseWithService (gh, targetTag) {
+  const previousTag = await gh.getPreviousTag(targetTag)
+  const commits = await gh.repo(previousTag, targetTag)
```

Comment:
Here you're calling the `repo` method from your `GitHubService` to get your commits, but this method only exists inside `Octokit`, so you could call it this way:
```suggestion
  const commits = await gh.gh.repo(previousTag, targetTag)
```
or even better use the `compareCommits` method you've implemented to retrieve the commits.
```suggestion
  const commits = await gh.compareCommits(previousTag, targetTag)
```

Reply:
Totally a bug, the intention was to call `compareCommits` anyway.

Reply:
Fixed on https://github.com/escaletech/circleci-gh-conventional-release/pull/5/commits/0db0c447c7109e274f2970cbe0678953f4091bf2.

I'll follow up on this PR with another one to improve testing.